Solutions/Menes.Abstractions/Menes/Internal/OpenApiDocumentProvider.cs

```diff
@@ -176,5 +190,31 @@ public bool GetOperationPathTemplate(string requestPath, string method, [NotNull
             operationPathTemplate = null;
             return false;
         }
+
+        private static OpenApiServer? MatchServer(Match match, string requestPath, OpenApiPathTemplate template)
+        {
+            string precursor = match.Index == 0 ? string.Empty : requestPath.Substring(0, match.Index);
+            if (template.PathItem.Servers.Count > 0)
+            {
+                return MatchPrecursor(precursor, template.PathItem.Servers);
+            }
+
+            if (template.Document?.Servers.Count > 0)
+            {
+                return MatchPrecursor(precursor, template.Document.Servers);
+            }
+
+            return null;
+        }
+
+        private static OpenApiServer? MatchPrecursor(string precursor, IList<OpenApiServer> servers)
+        {
+            return servers.FirstOrDefault(s =>
+            {
+                Regex regex = RegexCache.GetOrAdd(s, new Regex(UriTemplate.CreateMatchingRegex2(s.Url), RegexOptions.Compiled));
```

Comment:
I have been unable to work out why `UriTemplate` has both `CreateMatchingRegex` and `CreateMatchingRegex2`. I have been able to find only two differences at https://github.com/tavis-software/Tavis.UriTemplates/blob/master/src/UriTemplates/UriTemplate.cs#L384-L445

1. A comment after the first statement in the method (absent in the `2` version)
2. Meaninless whilespace in the `case "#":` body (absent in the `2` version)